server/connectors/slack_connector/client.py

```
@@ -26,33 +26,58 @@ def __init__(self, access_token: str):
+                    continue
+
+                response.raise_for_status()
+                return self._validate_response(response.json(), endpoint)
+                
+            except requests.RequestException as e:
+                if attempt < max_retries and "429" in str(e):
+                    retry_after = self._get_retry_delay(attempt)
+                    logger.warning(f"Rate limited on {endpoint}, retrying in {retry_after}s (attempt {attempt + 1}/{max_retries})")
+                    time.sleep(retry_after)
+                    continue
+                logger.error(f"Request to Slack API failed: {e}")
```

Ruff (0.15.15)

[warning] Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)